### DISCUSSION TITLE
Possible caching fix?

### DISCUSSION THREAD

**Discussion by alxspiker** [Mobile-Artificial-Intelligence/maid#60] [Ideas]
According to one of the web search results I found, you can use the clearTemporaryFiles() method from the file picker plugin to remove the cached files on Android. Alternatively, you can use a version of the file picker plugin below 2.0.0, which does not cache the files by default. However, this may also affect the functionality and compatibility of the plugin, so you should be careful when choosing the version. Another possible solution is to delete the cache directory manually before picking the files, as suggested by another web search result. However, this may also delete other files that are not related to the file picker, so you should be careful when doing this.

TL:DR - Use clearTemporaryFiles() method when starting a new convo and recache the model file again if nessecary. This might take longer to load but it would keep your storage from maxing out.

**Comment by alxspiker** [comment]
Basically when file picker picks a file, it caches it in the app files in case that file is deleted from your storage. So maybe even if you can just pull the path using filepicker and not actually use it for loading the file itself. Might require more cross platform code, IDK, trying to be helpful.

**Comment by alxspiker** [comment]
There's also a deleteOnExit() mathod that could be used to mark the file for deletion when the app closes.

If you want to delete a temporary file when the app is closed, you can use the `deleteOnExit()` method from the `File` class. This method marks the file to be deleted when the virtual machine terminates. For example, if you have a temporary file called `tempFile`, you can do something like this:

```dart
// Mark the temporary file to be deleted on exit
tempFile.deleteOnExit();
```

This way, you can ensure that the temporary file is deleted when the app is closed, regardless of how the app is closed. However, you should also be aware of the drawbacks of using the `deleteOnExit()` method. For example, this method may not work reliably on some platforms or file systems, and it may cause memory leaks if too many files are registered to be deleted.

**Comment by danemadsen** [comment]
Ill look into it, that caching issue youre referring to is an old one i think ive fixed now but i think theres a new one causing the app to crash.

**Reply by alxspiker** [reply]
I see what you mean, it seems to cause the caching issue on my android as soon as I select a model, not even run it yet which makes me think there is still a caching issue with the filepicker. I have to clear data and cache every few times I load a different model. I am constantly switching between multiple models mostly for research purposes, so it might be okay for someone only using 1 model, but it seems like every time I pick a new one it creates a new cached file, that and or its just caching every model file and not uncaching the old ones when I load a new one.

**Reply by danemadsen** [reply]
Yeah that's expected behavior unfortunately. See the flutter side of things doesn't actually need the file, just the file path because for generation llama.cpp handles loading the file. Ive tried various other flutter packages but none seem to be able to get the file path without caching
